vscode-dotnet-runtime-library/src/Utils/ExecutableArchitectureDetector.ts

```diff
@@ -0,0 +1,202 @@
+            }
+
+            return null;
+        }
+        catch (error)
+        {
+            return null;
+        }
+    }
+
+    public static IsKnownArchitecture(architecture: Architecture): boolean
+    {
+        return architecture !== null && architecture !== 'other';
+    }
+
+    /**
+     * Checks if the file is in ELF format
+     * Magic number: 0x7F 'ELF'
+     * Reference: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+     */
+    private isElfFormat(header: Buffer): boolean
+    {
+        return header[0] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.ELF.DELIMITER &&
+            header[1] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.ELF.E &&
+            header[2] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.ELF.L &&
+            header[3] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.ELF.F;
+    }
+
+    /**
+     * Checks if the file is in Mach-O format
+     * Magic number: 0xCF 0xFA 0xED 0xFE (Little-endian format)
+     * Reference: https://github.com/apple-oss-distributions/xnu/blob/main/EXTERNAL_HEADERS/mach-o/loader.h
+     */
+    private isMachOFormat(header: Buffer): boolean
+    {
+        return header[0] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.MACHO.CF &&
+            header[1] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.MACHO.FA &&
+            header[2] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.MACHO.ED &&
+            header[3] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.MACHO.FE;
+    }
+
+    /**
+     * Checks if the file is in PE/COFF format
+     * Magic number: 'MZ' (DOS header)
+     * Reference: https://learn.microsoft.com/en-us/windows/win32/debug/pe-format
+     */
+    private isPEFormat(header: Buffer): boolean
+    {
+        return header[0] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.PE.M &&
+            header[1] === ExecutableArchitectureDetector.EXECUTABLEFILESIGNATURES.PE.Z;
+    }
+
+    /**
+     * Parse ELF format architecture
+     * Reference: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+     * Specification: https://refspecs.linuxfoundation.org/elf/gabi4+/ch4.eheader.html
+     */
+    private getElfArchitecture(header: Buffer): Architecture
+    {
+        // e_machine field is at offset 0x12 (18)
+        const machine = header.readUInt16LE(0x12);
+
+        // Values from: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#Machine_codes
+        switch (machine)
+        {
+            case 0x03: return 'x86';      // EM_386
+            case 0x3E: return 'x64';      // EM_X86_64
+            case 0xB7: return 'arm64';    // EM_AARCH64
+            default: return 'other';
+        }
+    }
+
+    /**
+     * Parse Mach-O format architecture
+     * Reference: https://github.com/apple-oss-distributions/xnu/blob/main/EXTERNAL_HEADERS/mach-o/loader.h

```

Comment:
Reminded me of...
![image](https://github.com/user-attachments/assets/eaa6a537-4240-43bf-a7dc-fb44469c893c)

It is just anime mecha Batman, basically.

Reply:
It... totally looks like it. Glad you had fun reviewing this one 🤣